keyserver/testutil/testutil.go

```diff
@@ -126,3 +126,27 @@ func NewClient(t *testing.T, address string, msg []byte) ([]byte, error) {
 
 	return buf.Bytes(), nil
 }
+
+func NewBot(t *testing.T, msg []byte) ([]byte, error) {
```

Comment:
I see why you did this here, but you can split out the socket changes.

Maybe `NewTCPClient` and `NewUnixClient`.